### ISSUE TITLE
[idea] Supporting also `link` tags in `Response.follow`

### ISSUE THREAD

**Issue by starrify** [scrapy/scrapy#2785]
Currently the `Response.follow` method supports `Selector` objects whose root tag are `<a>`. It'll be nice to support also `link` tags.

Related code: [here](https://github.com/scrapy/scrapy/blob/fad6b70d/scrapy/http/response/text.py#L168)

Sample code:

```python
# Now
for href in response.css('link[rel=next]::attr(href)').extract():
    yield response.follow(href)

# When the proposed feature is available
for link_sel in response.css('link[rel=next]'):
    yield response.follow(link_sel)
```

**Comment by kmike** [comment]
I'm fine with such addition.
Do you have a real use case for it though?

**Comment by dangra** [comment]
implemented in #2789
